pontoon/sync/vcs/repositories.py

```diff
@@ -16,44 +16,55 @@ class PullFromRepositoryException(Exception):
 
 class PullFromRepository(object):
 
-    def __init__(self, source, target):
+    def __init__(self, source, target, branch):
         self.source = source
         self.target = target
+        self.branch = branch
 
     def pull(self, source=None, target=None):
         raise NotImplementedError
 
 
 class PullFromGit(PullFromRepository):
 
-    def pull(self, source=None, target=None):
+    def pull(self, source=None, target=None, branch=None):
         log.debug("Git: Update repository.")
 
         source = source or self.source
         target = target or self.target
+        branch = branch or self.branch
 
         command = ["git", "fetch", "--all"]
         execute(command, target)
 
+        remote = "origin"
+        if branch:
+            remote += "/" + branch
+
         # Undo local changes
-        command = ["git", "reset", "--hard", "origin"]
+        command = ["git", "reset", "--hard", remote]
         code, output, error = execute(command, target)
 
-        if code == 0:
-            log.debug("Git: Repository at " + source + " updated.")
-
-        else:
+        if code != 0:
```

Comment:
I'm not sure I understand this change. You will always print "Repo cloned" debug message, even if the repo is actually just updated.